core/App/screens/NameUpdate.tsx

```diff
@@ -0,0 +1,109 @@
+import { useNavigation } from '@react-navigation/core'
+import { StackNavigationProp } from '@react-navigation/stack'
+import React, { useContext, useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Keyboard, StyleSheet, Text, View } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import Icon from 'react-native-vector-icons/MaterialCommunityIcons'
+
+import Button, { ButtonType } from '../components/buttons/Button'
+import TextInput from '../components/inputs/TextInput'
+import AlertModal from '../components/modals/AlertModal'
+import Label from '../components/texts/Label'
+import { DispatchAction } from '../contexts/reducers/store'
+import { StoreContext } from '../contexts/store'
+import { useTheme } from '../contexts/theme'
+import { SettingStackParams, Screens } from '../types/navigators'
+import { testIdWithKey } from '../utils/testable'
+
+const NameUpdate: React.FC = () => {
+  const { t } = useTranslation()
+  const { ColorPallet } = useTheme()
+  const [state, dispatch] = useContext(StoreContext)
+  const [firstName, onChangeFirstName] = useState('')
```

Comment:
I would love to figure out if we could reduce the code duplication here with the other screen -- DRY. It's okay for today, but would be great to look in the future (especially with a merge to Bifold main)

Reply:
Possibly pass a navigation prop and conditionally render?